scripts/generate-dotnet.sh

```diff
@@ -1,14 +1,58 @@
 #!/bin/sh
 
-./scripts/install-tools.sh
+set -ex
+
+# TODO(kfcampbell): reduce duplication between this script and scripts/generate-go.sh
+if [ "$#" -lt 1 ] || [ "$#" -gt 2 ]; then
```

Comment:
It may be smart to refactor this type of logic into some kind of tools script, and then import functions between scripts. Perhaps that's a job for a future PR?

Reply:
I would hold off on abstraction until all of the dust settles from all of the changes for the enterprise effort... my guess is that we'll find other areas of optimization as well.